### ISSUE TITLE
Documentation cleanup

### ISSUE THREAD

**Issue by bruchar1** [mesonbuild/meson#11475]
I think there could be some improvements to the way the documentation is currently organized.

- There is duplicated information about how to install Meson in **The Absolute Beginner's Guide**, **Getting Meson**, and **Quickstart Guide**.
- Tutorials could be grouped in a common section: ### Tutorial, **Building a simple SDL2 app**, **An in-depth tutorial**
- Most sections having subsections are simply a list of the subpages. It is a good thing because we can easily skip them by expanding it and clicking directly on a subsection. However, **Getting Meson**, **Contributing to Meson**, and **Wrap dependency system manual** are pages having both contents and subsections. It is easy to miss the contents of those pages.
- I'm not sure if it worth keeping the translated versions of the **Getting Meson** page. 
- **Meson file rewriter** should probably go into the **Manual** section
- The **Manual** section is becoming big. It should probably be reorganized as well.
- There could be a subsection in **Manual** about specific languages: **Java**, **Vala**, **D**, **Cython**, **Rust**, etc.
- There is duplication between **Visual Studio's external build projects** and **Using with Visual Studio**. Those pages should be referenced somehow in **IDE integration** page
- **Users**, **In the press** and **Conference presentations** could probably be regrouped in a common section
- **playground** page could be removed
- **FAQ** and **How do I do X in Meson** could probably be merged together

I (or somebody else) could probably open a PR for that, but before beginning, I want to be sure everybody agree on the plan. Also, is there any other desirable changes I missed?

**Comment by bruchar1** [comment]
Could also include: #11386

**Comment by rgommers** [comment]
I'll note that an in-depth review of the Meson docs was posted by @CAM-Gerlach in https://github.com/mesonbuild/meson-python/issues/197#issuecomment-1312712210, on request of @eli-schwartz. I'll repost that in full here: 

@eli-schwartz My first high-level impression, as both a first-time visitor to the site with minimal Meson experience and as a documentarian, is that the docs look quite complete and comprehensive, with all the major pieces there (tutorials, how-tos, user reference, API reference, some amount of explanations, and a healthy helping of ancillary, meta- and developer-focused docs. However, at least to me, the current organization and structure of the docs feels a little overwhelming, not so well-organized, hard to navigate easily and logically and potentially confusing.

<details>

<summary>Specific first impressions</summary>

For example, as a new user starting out, there's a "Getting Meson" guide, an "Absolute Beginners Guide to Installing and Using", a "Quickstart Guide", a Tutorial, what appears to be another tutorial with a very specific name (building SDL2 app), all at the top level—and under Additional Documentation, there's _another_ In-Depth Tutorial. Where should I start, and where should I go from there? Most of these give some kind of introduction, but I don't see a lot connections between these documents (now that you've done XXX, you can move on to YYY and ZZZ in order to...). It could use a coherent organizational structure, and a high-level page/section listing these learning resources in one place, and at least briefly introducing each and providing a suggested sequence.

Likewise, I went looking for the "API reference docs" you referred to above, and was a bit confounded by the fact that there's a Manual, a Reference Manual, Reference Tables, and a Quick Reference menu, all at the top level. Peering inside the "Manual", I was hit with a massive flat list of topics, from which I quickly retreated. After a bit of digging and looking at things, I figured out you must be referring to the sections in the Reference Manual. Indeed, seems clear enough—within that section, it's well organized, and suitably formatted and structured for API reference docs. I was a bit confused at first when the topics I was seeing at the top of the top-level Reference Manual page didn't match those in the sidebar, though I did figure out it was just in a different order, for some reason. Searching for methods, even by fully qualified name, was also not so friendly or easy to navigate, and didn't always work reliably, though I assume that has more to do with your docs system (I haven't heard of `hotdoc` before) than anything.

Aside from that, if I were looking for how to do something specific with Meson (i..e. a "How to" guide), there's the How do I do X in Meson with a long flat list of tasks, but there's also a number of misclanious how-to-type documents floating around in the root, others under "Additional documentation" as well as some "How to" type material in the FAQ. Likewise, I'd expect to find most of the developer documentation and meta-documents under "Contributing to Meson", but some of that is under Additional Documentation too. And overall, the ordering did not seem the most logical.

</details>

I could provide more detailed recommendations for a more helpful, logical and organized structure here or in a more appropriate venue, but overall, I'd suggest:

* Greatly reduce the number of top-level items to just those that serve common and distinct user needs, and in a logical order, You don't have to strictly follow Diataxis, but it provides a good starting point—Tutorials [for learning], How-to guides [for tasks], Reference [for information], and Explanation [for background material], plus developer/contributing/meta docs, a FAQ and maybe a _one_ installing/quickstart guide.
* Likewise, avoid having top-level sections that have a very lengthy flat list of documents directly underneath them, especially without a clear and unambiguous order to them. Instead, move these into logical subsections as much as practical.
* For each top-level section, provide a landing page that gives a brief overview of the major pages/subsections within, in a manner appropriate to that user need—for example, for the tutorials, you might want to clarify which audience each tutorial/etc. is aimed at, and a suggested progression between each.
* Within prominient first and second-level docs, especially things like tutorials, link related resources at the top, and next steps at the bottom, to help the reader navigate the docs and feel a sense of coherence and direction.
* Avoid standalone or overly specific docs floating around in the root; they just clutter things up and could typically go well under a more appropriate subsection.

Again, all the moving pieces look to more or less be there—compared to all the work that's clearly gone into them so far, a very modest amount of thoughtful organization should help go a long way toward making all the detailed content you have friendlier to new users, quicker to navigate for experienced users, and more accessible, discoverable and useful to everyone in between.

**Comment by eli-schwartz** [comment]
Yup, still trying to find time to ponder documentation rewrites :P that feedback is great, assuming I ever get around to implementing it.

If someone else would like to do the work for that, even better! I really should have cross-posted that myself, in retrospect.

**Comment by tintou** [comment]
This is a first step on making the documentation more manageable https://github.com/mesonbuild/meson/pull/15894
